modules/features2d/include/opencv2/features2d.hpp

```diff
@@ -107,6 +107,11 @@ class CV_EXPORTS KeyPointsFilter
      * Remove keypoints from some image by mask for pixels of this image.
      */
     static void runByPixelsMask( std::vector<KeyPoint>& keypoints, const Mat& mask );
+    /*
+     * Remove objects from a vector based on keypoints and mask.
+     */
+    template <typename T>
+    static void runByPixelsMask2(std::vector<KeyPoint> &keypoints, std::vector<T> &removeFrom, const Mat &mask);
```

Comment:
exposing template function API without providing implementation in the header is a bug. Can we move this function into .cpp completely?